server/src/main/java/org/elasticsearch/cluster/metadata/ComposableIndexTemplate.java

```diff
@@ -140,34 +152,12 @@ public ComposableIndexTemplate(
         @Nullable DataStreamTemplate dataStreamTemplate,
         @Nullable Boolean allowAutoCreate
     ) {
-        this(indexPatterns, template, componentTemplates, priority, version, metadata, dataStreamTemplate, allowAutoCreate, null);
-    }
-
-    ComposableIndexTemplate(
```

Comment:
I've removed this constructor as it only had 4 callers. As it was already package-protected, there's a low risk that this could break serverless. I've also manually checked that the constructor wasn't used there.

Reply:
TIL!